GOVERNANCE.md

```diff
@@ -0,0 +1,458 @@
+# Governance thin-edge.io
+
+This documents serves as a guiding document to ease technical decision-making in the project as a whole.
+
+Some of its goals are:
+
+- Making sure that the different stakeholders are represented with their requirements
+- Making sure that contributors know how decisions are made and who they can address concerns/questions to
+- Making sure that team members know what is required of them for the project
+
+
+## The project vision
+
+In the [vision.md](./vision.md) the goal of the project is encoded.
+It specifies the different functional requirements and non-functional requirements of the project as a whole.
+All the stakeholders must agree on the vision upon joining.
+
+- The current stakeholders are: Software AG, and IFM
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issues in their assigned area, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
+* The teams follow the project hierarchy for decisions, i.e. higher decisions take precedence
+
+The important bits:
+
+At the 'top' are the maintainers, whose job is to define and realize the project vision of thin-edge.io.
+Maintainers should strive for agreement based on the following factors:
+
+- The vision of the whole project
+- Feasibility, accounting for future technical and cultural shifts
+- The health of the project going forward
+
+Overall, the maintainers should not be beholden to their respective companies, and instead to the health of the project and its community.
+
+Underneath the maintainers are the teams or individuals, that each are allowed to  make similar decisions about the project *in the area they have been delegated*.
+
+
+- Some decisions are global in the project, like the core/API
+    - Team members should try to make sure to explore possible options together before calling for time-consuming solutions like a vote. This mostly includes uncontroversial changes that are low impact or have already been agreed upon
+    - If an exclusive choice has to be made (where it is not possible to entertain two conflicting approaches), and no clear side is more advantageous to pick, a vote should be held with the majority opinion being the final decision.
+
+- Some decisions are local, e.g. a plugin does not impact others with its behaviour, however it still needs to be a 'good citizen' in the project (CI, format, etc..)
+    - These should be clearly scoped in their respective project part
+    - However, if needed a 'higher up' decision can be requested if no consensus is achieved
+
+
+
+## Project Structure
```

Comment:
I am not sure what steering committee you are referring to? The document already contains a delineating of the two major ‚roles‘ in the context of this document.

-> Maintainers are responsible and get to decide
-> everyone else is a contributor


Maybe it can be made clearer in the document, but it is vital that nobody gets to make „decisions“ that is not a de facto contributor of the project. Hence no committee or similar.

Reply:
in the sections above we talk about "core contributors = [SAG, IFM]"
"Project leads, one each from [core contributors]" , we should be very explicit and consistent with the naming and agree with your points on decisions. I mentioned one point multiple times that somehow should be captured: There is company investment here, so  if decisions from a company are made to pull out or invest into contribution (e.g. hiring more people) in similar areas regarding the "core" it should be transparent. There is much more then just technical decisions e.g. onboarding new partners, community work/events, promotion , tradefairs etc. If the should align with the project "health" it should be open and transparent and there might be a group we could form around that aswell which doesn't meet the term contributor or maintainer. 

--> For now I am happy to push this into a later version of the gov model

Reply:
I'm aligned with the @TheNeikos here. The product-related decision makers of the OSS project must be only active contributors. I think that investment-related decisions are to be taken at a different level, independent of this governance model. A partner might decide to invest more or less in this project (by using it, contributing to it, or hiring for it), but along its own decision process with no ruled coordination with other partners. The point of this governance model it to ensure that these investments can be long-term, with a clear and stable vision (we plan and commit to go there) as well as guaranties that independent partners can contribute in harmony despite different motivations (what is built on top of thin-edge by one must not be broken by another).

Reply:
We agreed that after the governance model is in place we need to review the current power dynamics and decision-making prosses in the involved companies.